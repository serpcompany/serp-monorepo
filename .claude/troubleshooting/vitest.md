https://vitest.dev/guide/common-errors.html