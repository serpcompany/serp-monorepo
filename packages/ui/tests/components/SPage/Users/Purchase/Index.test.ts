import { describe, expect, it, vi } from 'vitest';
import ComponentRender from '../../../../componentRender';
import PurchaseIndex from '../../../../../components/SPage/Users/Purchase/Index.vue';
import { mockNuxtImport } from '@nuxt/test-utils/runtime';

// Global route mock so we can simulate different query parameters
let routeMock = { query: {} };
globalThis.useRoute = () => routeMock;

// Global router mock with a spy on push
const routerMock = { push: vi.fn() };
globalThis.useRouter = () => routerMock;

describe('SPage/Users/Purchase/Index Snapshot', () => {
  const scenarios: [string, { query: Record<string, unknown> }][] = [
    [
      'renders success state correctly',
      { query: { success: 'true', redirectTo: '/success' } }
    ],
    [
      'renders cancelled state correctly',
      { query: { cancel: 'true', redirectTo: '/cancel' } }
    ],
    [
      'renders error state correctly',
      { query: { error: 'Payment failed', redirectTo: '/error' } }
    ],
    ['renders unknown state correctly', { query: {} }]
  ];

  it.each(scenarios)('%s', async (desc, { query }) => {
    // Update our global route mock for the current scenario
    routeMock = { query };
    const html = await ComponentRender(
      `SPage/Users/Purchase/Index ${desc}`,
      {},
      PurchaseIndex
    );
    expect(html).toMatchSnapshot();
  });
});
