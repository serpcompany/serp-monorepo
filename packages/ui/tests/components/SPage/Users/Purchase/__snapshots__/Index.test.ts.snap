// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPage/Users/Purchase/Index Snapshot > renders cancelled state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❓ Unknown</h1>
    <p>Unknown error.</p>
  </div>
  <p> Redirecting to <a href="#">/</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase/Index Snapshot > renders error state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❓ Unknown</h1>
    <p>Unknown error.</p>
  </div>
  <p> Redirecting to <a href="#">/</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase/Index Snapshot > renders success state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❓ Unknown</h1>
    <p>Unknown error.</p>
  </div>
  <p> Redirecting to <a href="#">/</a> in 10 seconds. </p>
</div>"
`;

exports[`SPage/Users/Purchase/Index Snapshot > renders unknown state correctly 1`] = `
"<div class="success-container">
  <div>
    <h1>❓ Unknown</h1>
    <p>Unknown error.</p>
  </div>
  <p> Redirecting to <a href="#">/</a> in 10 seconds. </p>
</div>"
`;
